File: Febuary/squares.cpp
#include <iostream>
#include <string>
#include <algorithm>
#include <cctype>
#include <vector>
#include <cmath>
using namespace std;

//logic: 
// 1. if number is negative, flip to positive -> 
// 2. square the number -> 
// 3. append the squared number to the array -> 
// 4. sort the array
// 5. return the array


vector<int> sortedSquares(vector<int>& nums) {

    vector<int> squares = {};

    for (int number : nums) {
            squares.push_back(number * number);
    }

    sort(begin(squares), end(squares)); // sort after the all numbers have been added to squares
    return squares; // return the array 
}


int main(){
    vector<int> nums = {-4,-1,0,3,10};

    vector<int> result = sortedSquares(nums);

    for (int num : result){
        cout << num << " ";
    }

    cout << endl;
    return 0;
}

File: Febuary/squares.py
'''
Given an integer array nums sorted in non-decreasing order, return an array of the squares of each number sorted in non-decreasing order.
'''
import math

#logic: 
# 1. if number is negative, flip to positive -> 
# 2. square the number -> 
# 3. append the squared number to the array -> 
# 4. sort the array
# 5. return the array

def sortedSquares(nums: list[int]) -> list[int]:
    squared = []

    for number in nums:
        if number < 0:
            number = abs(number)
        
        number = number * number
        squared.append(number)

    squared.sort()
    
    return squared


if __name__ == '__main__':
    nums = [-4,-1,0,3,10]
    print(sortedSquares(nums=nums))